python/API_GW_CACHE_ENABLED_AND_ENCRYPTED/API_GW_CACHE_ENABLED_AND_ENCRYPTED.py

```diff
@@ -0,0 +1,404 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+Description:
+  Checks whether the API GW stage has cache enabled and is encrypted. The rule is NON_COMPLIANT if the API GW stage is not configured for cache and is not encrypted.
+
+Trigger:
+  Configuration Change
+
+Reports on:
+  AWS::ApiGateway::Stage
+
+Rule Parameters:
+  None
+
+Scenarios:
+  Scenario: 1
+     Given: Cache is not enabled for the API Gateway Stage
+      Then: Return NON_COMPLIANT
+  Scenario: 2
+     Given: Cache is enabled for the API Gateway Stage
+       And: Cache is not enabled for one or more methods in the API Gateway Stage
+      Then: Return NON_COMPLIANT
+  Scenario: 3

```

Comment:
Scenario 3 feels like a duplicate because `Cache is not enabled for one or more methods in the API Gateway Stage` is already covered in Scenario 2. And the rest is covered in Scenario 4.